### PR TITLE
feat: Space CRD

### PR DESCRIPTION
**Pull request by xcoulon**
see also https://github.com/codeready-toolchain/api/pull/267

Fixes https://issues.redhat.com/browse/CRT-1309

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,4 +31,6 @@ require (
 	sigs.k8s.io/controller-runtime v0.8.3
 )
 
+replace github.com/codeready-toolchain/api => github.com/xcoulon/api v0.0.0-20211123162731-1ef4b458d4e1
```
Comment (rajivnathan) [inline go.mod:34]: Just for reminder purposes 😃

**Review by MatousJobanek — COMMENTED** [review]
This PR is missing a few important changes.
you need to:
1. modify the kustomize config files (config/crd/kustomization.yaml)
2. define this new resource in the PROJECT file
3. add the CRD to the kustomized CSV file config/manifests/bases/host-operator.clusterserviceversion.yaml

**Comment by xcoulon** [comment]
> This PR is missing a few important changes. you need to:
> 
> 1. modify the kustomize config files (config/crd/kustomization.yaml)
> 2. define this new resource in the PROJECT file
> 3. add the CRD to the kustomized CSV file config/manifests/bases/host-operator.clusterserviceversion.yaml

done in 5b077b1 (hopefully)

File: go.mod [hunk 2]
```diff
@@ -31,4 +31,6 @@ require (
 	sigs.k8s.io/controller-runtime v0.8.3
 )
 
+replace github.com/codeready-toolchain/api => github.com/xcoulon/api v0.0.0-20211125125708-0f5ec2774aa7
```
Comment (MatousJobanek) [inline go.mod:34]: reminder

**Review by MatousJobanek — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (xcoulon) [reply go.mod]: taken care of in https://github.com/codeready-toolchain/host-operator/pull/544/commits/deae7ae645802efceabdad9ff7938e97e6a3d2d8

**Pull request merged**